subprojects/logging/src/main/java/org/gradle/internal/featurelifecycle/LoggingDeprecatedFeatureHandler.java

```diff
@@ -83,13 +84,13 @@ public void featureUsed(DeprecatedFeatureUsage usage) {
     }
 
     private void fireDeprecatedUsageBuildOperationProgress(DeprecatedFeatureUsage usage) {
-        if (buildOperationProgressBroadcaster != null) {
-            buildOperationProgressBroadcaster.progress(usage);
+        if (progressEventEmitter != null) {
+            progressEventEmitter.emitNowIfCurrent(new DefaultDeprecatedUsageProgressDetails(usage));
```

Comment:
If this is the only place where the `ifCurrent()` set of methods are used, I think it would be better to inline that functionality here. The code here would be more descriptive ("if current" doesn't really explain to me what is happening), and the code in the emitter would be simpler.

Reply:
Let's hash this out in a follow up.